git/util.py

```diff
@@ -116,14 +109,33 @@
 
 log = logging.getLogger(__name__)
 
-# types############################################################
+
+def _read_env_flag(name: str, default: bool) -> bool:
+    try:
+        value = os.environ[name]
+    except KeyError:
+        return default
+
+    log.warning(
+        "The %s environment variable is deprecated. Its effect has never been documented and changes without warning.",
```

Comment:
In the case of `HIDE_WINDOWS_FREEZE_ERRORS`, even the attribute itself is nowhere documented and it's not listed in `__all__`, so to say that anything about it is deprecated is actually an understatement: it has never been part of the public interface of any component of GitPython and still isn't. It seems to me that users will not wrongly start believing it is okay to use or rely on something *because* of a claim that it is deprecated. But I could try to think of more precise yet still intuitively clear wording, if you think this is an issue.

Reply:
Thanks for playing it so very safe, it's appreciated.